q2_fmt/_peds.py

```diff
@@ -468,3 +484,89 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
 def _mask_recipient(donor_mask, recip_df):
     maskedrecip = donor_mask & recip_df
     return maskedrecip
+
+
+def peds_bootstrap(table: pd.DataFrame, metadata: qiime2.Metadata,
+                   time_column: str, reference_column: str,
+                   subject_column: str,
+                   filter_missing_references: bool = False,
+                   drop_incomplete_subjects: bool = False,
+                   drop_incomplete_timepoint: list = None,
+                   bootstrap_replicates: int = 999) -> (pd.DataFrame):
+
+    metadata_df = metadata.to_dataframe()
+    donor = metadata_df[metadata_df.index.isin(
+        _check_reference_column(metadata=metadata_df,
+                                reference_column=reference_column))]
+
+    recipient = metadata_df.loc[metadata_df[reference_column].notnull()]
+    fake_donor = pd.DataFrame([])
+    for i in range(0, bootstrap_replicates+1):
+        if i == 0:
+            peds = sample_peds(
+                table=table, metadata=metadata,
+                time_column=time_column,
+                reference_column=reference_column,
+                subject_column=subject_column,
+                filter_missing_references=filter_missing_references,
+                drop_incomplete_subjects=drop_incomplete_subjects,
+                drop_incomplete_timepoint=drop_incomplete_timepoint)
+            real_temp = peds["measure"]
+        else:
+            shifted_list = recipient[reference_column].sample(frac=1).to_list()
+            recipient.loc[:, reference_column] = shifted_list
+            metadata_df = pd.concat([donor, recipient])
+            metadata = qiime2.Metadata(metadata_df)
+            peds = sample_peds(
+                table=table, metadata=metadata,
+                time_column=time_column,
+                reference_column=reference_column,
+                subject_column=subject_column,
+                filter_missing_references=filter_missing_references,
+                drop_incomplete_subjects=drop_incomplete_subjects,
+                drop_incomplete_timepoint=drop_incomplete_timepoint)
+            fake_donor[i] = peds["measure"]
+    fake_donor_series = fake_donor.median(axis=1)
+    # Common Langauge effect size calcs in prep for stats refactor.
+    # These are not being used currently but will be soon
+    fake_donors_CLES = fake_donor.median(axis=0)
+    agree = ((fake_donors_CLES <
+              real_temp.median()).sum())/bootstrap_replicates
+    disagree = 1 - agree
+    cles = agree - disagree
+    cles
```

Comment:
Looks like a debugging statement?

Reply:
And are you using this value again anywhere? I don't see it.

Reply:
I asked @cherman2 to calculate the common language effect size (which is the `agree` variable here) and the rank biserial correlation (which is actually `cles`) while the bootstrap was still fresh. It will ultimately end up in a stats-table, so we will just add these two values as new columns.

I think the solo expression on the single line can definitely be removed.

Reply:
@ebolyen Currently these aren't being added to the stats table (because we didn't seen them) So they are currently just un-used variables (hence the solo expression so flake8 wouldn't yell at me). 

I will add these into the table in this review so that they aren't just un-used variables and then I will remove the solo expression.

Reply:
These have now bee added to the stats table.